File: src/types/drive.ts
export type DriveType = "SSD" | "HDD" | "Removable" | "Network" | "Unknown";

export interface DriveInfo {
    name: string;
    mount_point: string;
    total_bytes: number;
    free_bytes: number;
    used_bytes: number;
    drive_type: DriveType;
    file_system: string;
}

File: src-tauri/src/models/drive.rs
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
pub struct DriveInfo {
    pub name: String,
    pub mount_point: String,
    pub total_bytes: u64,
    pub free_bytes: u64,
    pub used_bytes: u64,
    pub drive_type: DriveType,
    pub file_system: String,
}

#[derive(Debug, Clone, Serialize)]
pub enum DriveType {
    SSD,
    HDD,
    Removable,
    Network,
    Unknown,
}
